### ISSUE TITLE
Add support for new renderImage and renderVideo dataSources

### ISSUE THREAD

**Issue by bryanbraun** [bryanbraun/checkboxland#14]
The `renderImage` and `renderVideo` plugins would be a bit more flexible if they accepted other `dataSources` like:
* UInt8ClampedArray
* ArrayBuffer
    * We would need to convert it into a UInt8ClampedArray before we could put it on our canvas (see https://jameshfisher.com/2020/03/01/how-to-write-an-arraybuffer-to-a-canvas/).

These dataSources would add support for additional use cases, including:
* Uploading longer videos (those that are too long to be successfully converted to a data URL).
* Enable uploading of animated gifs, using third-party conversion tools like [Gifler](https://themadcreator.github.io/gifler/) or [Gifuct](http://matt-way.github.io/gifuct-js).
* ...probably other ones!

**Comment by KonradLinkowski** [comment]
Hi @bryanbraun is it still an issue? I can look into this if so :)

**Comment by bryanbraun** [comment]
Yes, @KonradLinkowski, it's still available if you want to give it a shot!
